src/libstd/sys/windows/fs.rs

```diff
@@ -333,6 +334,90 @@ impl File {
         Ok(newpos as u64)
     }
 
+    pub fn remove(&self) -> io::Result<()> {
+        let mut info = c::FILE_DISPOSITION_INFO {
+            DeleteFile: -1,
+        };
+        let size = mem::size_of_val(&info);
+        try!(cvt(unsafe {
+            c::SetFileInformationByHandle(self.handle.raw(),
+                                          c::FileDispositionInfo,
+                                          &mut info as *mut _ as *mut _,
+                                          size as c::DWORD)
+        }));
+        Ok(())
+    }
+
+    pub fn rename(&self, new: &Path, replace: bool) -> io::Result<()> {
+        // &self must be opened with DELETE permission
+        use iter;
+        #[cfg(target_arch = "x86")]
+        const STRUCT_SIZE: usize = 12;
+        #[cfg(target_arch = "x86_64")]
+        const STRUCT_SIZE: usize = 20;
+
+        // FIXME: check for internal NULs in 'new'
```

Comment:
This `to_u16s` helper may be useful here, although it may need some tweaks if you want to avoid the extra allocation.